### ISSUE TITLE
[BUG] Intelli-Sense Stops Working In Every Language

### ISSUE THREAD

**Issue by angelhdz** [microsoft/vscode#80289]
VSCode Version: **1.37.1**
OS:  **Windows 10 Pro 64bit**

Intelisense Stops Working In Every Language. I'm tired of having to Reload the workspaces to get it working again.

**Comment by sbatten** [comment]
@angelhdz can you provide more information on what you mean? For example, which languages are you working with and extensions you have installed?

**Comment by angelhdz** [comment]
@sbatten of course! Thanks for replying.

**Python** with following extensions: 

- ms-python.python
- donjayamanne.python-extension-pack
- seanwu.vscode-qt-for-python
- batisteo.vscode-django
- wholroyd.jinja

**Actionscript** with following extensions:

- bowlerhatllc.vscode-nextgenas
- dbaeumer.vscode-eslint
- vshaxe.haxe-debug

**Javascript** and **NodeJS** with following extensions:

- dbaeumer.vscode-eslint

And not language related extensions like:

- coenraads.bracket-pair-colorizer-2
- sleistner.vscode-fileutils

**Comment by sbatten** [comment]
Are you receiving any error notification that indicates that the extension host is unresponsive? Can you also check dev tools (Ctrl+Shift+i) for any related errors?

**Comment by angelhdz** [comment]
Nope. No errors. It just stops auto-completing everything after a while of working and debugging. Then I reload the workspace and it gets fixed.  Also, sometimes, I get errors like "undefined property csbedfasp" because it detected something I wrote recently as If I was reading a property. Then I save the current file, and the errors goes away. It's like Intelli-Sense just stops working.

**Comment by sbatten** [comment]
@angelhdz can you try running with extensions disabled and see if the issue reproduces? If it does not, we might need to enable them one by one to figure out the culprit.

**Comment by angelhdz** [comment]
Sure, I can do that, but if I disable for example **Python > ms-python.python**  extension in the Python workspace, I can't debug or have python code hinting.

What I'm gonna do is disable **Python Extension Pack > donjayamanne.python-extension-pack** and see if that extension is in conflict with the other one.

**Comment by angelhdz** [comment]
Hmmm... this makes sense. I think  **donjayamanne.python-extension-pack** extension is in conflict with **ms-python.python** extension. Both of them have code hinting/auto complete, but they display them differently in the drop down window. So I will raise an Issue in their repository.

Question, is there any way to tell one extension not to do code hinting and let another extension controls the code hinting?

**Comment by sbatten** [comment]
If disabling that extension pack helps, I believe you have narrowed it down to one of the extensions inside of that pack. I don't think that pack conflicts with the ms-python extension since it contains that extension within it. If just ms-python works, then then there are 4 remaining extensions. Can you disable them individually to see which is causing the issue?

![image](https://user-images.githubusercontent.com/6561887/64215207-78326500-ce68-11e9-8180-d83b08045330.png)

**Comment by angelhdz** [comment]
Yup. **donjayamanne.python-extension-pack** is the one. It stops  doing code-hinting after a while of working with it. 

This a code hinting with only VSCode Intelli-Sense:
![image](https://user-images.githubusercontent.com/10191085/64216262-184ab700-ce86-11e9-9bc7-21990ceb7101.png)

This a code hinting with the extension:

![image](https://user-images.githubusercontent.com/10191085/64216316-5d6ee900-ce86-11e9-9e8f-7502d13b2c4d.png)

And that is the code hinting that stops working.

### Update

 I just noticed that if I disable the **Visual Studio IntelliCode** extension that comes with that extension package, I still have the good looking code hinting, so I will test with this extension disabled for a while to test.

Thank you for your time!

**Comment by sbatten** [comment]
@angelhdz good to know, is you notice that the issue remains resolved with Intellicode disabled, please file an issue on their repository at https://github.com/MicrosoftDocs/intellicode/issues

**Comment by angelhdz** [comment]
It happened again with **Visual Studio IntelliCode** extension disabled. So, the issue must be with **MagicPython** extension instead. And I think this makes sense, because in the Magic Python extension folder, I edited some *.json files to add the word "this" as a keyword (to replace 'self' keyword) and it worked, the word 'this' became color highlighted. So this extension is responsible of intelli-sense/code hinting and color highlighting, the decorations in the code-hinting auto complete drop down menu, and other Python related actions I'm not aware of.

**Comment by angelhdz** [comment]
### Update

I uninstalled all Python related extensions, but this one **ms-python.python** (obviously, I need that one to debug Python files and have python language server, etc.).

I noticed in the status bar "Analizing In Background 800> items*" and **while that was running, I couldn't have intelli-sense/code hinting**. 

So I researched a bit, and some answers suggesting disabling Jedi. So I set **python.jediEnabled** to false and Reloaded the workspace. 

Will test for a while to see what I find now. Thank you.

**Comment by angelhdz** [comment]
### Final Words

I can finally certify **Magic Python** extension is the one conflicting with  **ms-python.python**'s Intelli-Sense.

When I'm using only **ms-python.python**, everything is just fine. When I install **Magic Python**, either as stand-alone or with the **donjayamanne.python-extension-pack**, after a while of running, it's Intelli-Sense just stops working and it doesn't code-hint anymore until I reload the workspace.

So that means, In every language I work with, if I have issues with Intelli-Sense, it's the developer's extension's fault? How to know when maybe VSCode is at fault?

**Comment by sbatten** [comment]
@angelhdz unfortunately extensions can be bad actors and not play nicely with others. In this case it seems to be *Magic Python* causing issues and a bug should be reported against them. In the future, a similar process should be followed where extensions are disabled to see if they are the cause. Even if an extension is causing the issue, it may turn out to be a VS Code issue, but typically that's easiest for the extension owner to discover once they know the issue.

**Comment by angelhdz** [comment]
### Thanks

Now this issue is closed for good.

**Comment by angelhdz** [comment]
OMG... issues again!

I only have the Python extension **ms-python.python** the one that runs the debugger, and runs the python linter and language server. 

After a while of working, it stops auto completing things, then I reload the workspace and I get code-hinting again. Also, sometimes I get errors in the **Problems** tab, that really don't exist, because without changing any code, I reload the workspace and those errors go away. 

This has me thinking... maybe the extension stops working? Maybe VSCode's Intellisense stops working?

**Now that I remember**, this happens to me also in the Actionscript 3.0 Language workspace that I created, with its own extension **Actionscript & MXML** and I remember I contacted the developer and explained this and he couldn't help me, so my only solution was to restart the Actionscript Language Server or reload the workspace.

For example, the errors can be "Undefined property "asdfd" and I don't have any "asdfd" property in my code. It's like while I'm typing on the editor, the intelli-sense stops working and messes everything up.
